### DISCUSSION TITLE
Where is the 'jre/' directory for OpenJDK?

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#1026] [Everyday usage] [answered]
I installed the OpenJDk by brew, but I don't find the `jre/` directory in `/opt/homebrew/Cellar/openjdk/15.0.2/libexec/openjdk.jdk/Contents/Home`.

Actually, I want to use the jre.jar, rt.jar, and classes.jar for Soot.

Does anyone know? Thank you.

**Comment by alebcay** [comment, marked as answer]
According to [JEP 220](https://openjdk.java.net/jeps/220), `rt.jar` and some other internal structures of the JDK/JRE have changed. Starting with Java 9, most of the stuff in those JAR files have been moved and restructured in the JMOD format, and are located in `#{prefix}/libexec/jmods`.

You probably need a newer version of Soot that supports Java 9+ and is able to work with the new JMODs rather than the old JAR format.

**Reply by ghost** [reply]
Thank you, it helps me a lot.

**Comment by carlocab** [comment]
You may also want to use a stable path instead of one that refers to the Cellar. You can find this with
```
$(brew --prefix openjdk)/libexec/openjdk.jdk/Contents/Home/jmods
```
for you, this should return
```
/opt/homebrew/opt/openjdk/libexec/openjdk.jdk/Contents/Home/jmods
```

**Reply by ghost** [reply]
Thank you, it helps me a lot.
